domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -170,17 +175,16 @@ public async Task<ActionResult<EditTransferAgreementEndDate>> EditEndDate(Guid i
     }
 
     [ProducesResponseType(typeof(string), 200)]
+    [ProducesResponseType(typeof(string), 500)]
     [HttpPost("wallet-deposit-endpoint")]
     public async Task<ActionResult> CreateWalletDepositEndpoint()
     {
-        var issuer = User.FindIssuerClaim();
-        var audience = User.FindAudienceClaim();
-        var subject = User.FindSubjectGuidClaim();
-        var name = User.FindSubjectNameClaim();
-
-        var base64String = await walletDepositEndpointService.CreateWalletDepositWithToken(new JwtToken(issuer, audience, subject, name));
-
-        return Ok(new { result = base64String });
+        if (AuthenticationHeaderValue.TryParse(httpContextAccessor.HttpContext?.Request.Headers["Authorization"], out var authentication))
+        {
+            var base64String = await walletDepositEndpointService.CreateWalletDepositWithToken(authentication.Parameter);
```

Comment:
Passing parameters all the way down to the services often leads to hard to read code. Have you considered the option to get the GRPC client from DI? See https://learn.microsoft.com/en-us/aspnet/core/grpc/clientfactory?view=aspnetcore-7.0#call-credentials

Or maybe create a service to be used by `WalletDepositEndpointService`, which is capable of getting the JWT token.

Reply:
A good point. I have actually already made a tech debt issue, on the zenhub board, outlying a need for some refactoring. right now there is a lot of business logic in the controllers, which can be moved to their own respective services, in a service layer. If it is fine with you, I would like to handle it, there?

Reply:
ok for me